tokio-fs/src/file/metadata.rs

```diff
@@ -0,0 +1,39 @@
+use super::File;
+
+use futures::{Future, Poll};
+
+use std::fs::File as StdFile;
+use std::fs::Metadata;
+use std::io;
+
+const POLL_AFTER_RESOLVE: &str = "Cannot poll MetadataFuture after it resolves";
+
+/// Future returned by `File::metadata` and resolves to a `(Metadata, File)` instance.
+#[derive(Debug)]
+pub struct MetadataFuture {
+    file: Option<File>,
+}
+
+impl MetadataFuture {
```

Comment:
I would remove this entire impl block and just keep the `metadata` method on `File`.

Reply:
I'm following the pattern of having a `new` started by `CreateFuture` and `OpenFuture`. This also provides a nice place to put the annoying `as_mut` / `expect` / `std` delegation.